src/main/java/school/hei/haapi/service/FileInfoService.java

```diff
@@ -0,0 +1,40 @@
+package school.hei.haapi.service;
+
+import static school.hei.haapi.service.aws.FileService.getFormattedBucketKey;
+
+import java.time.Instant;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import school.hei.haapi.endpoint.rest.model.FileType;
+import school.hei.haapi.model.FileInfo;
+import school.hei.haapi.model.User;
+import school.hei.haapi.repository.FileInfoRepository;
+
+@Service
+@AllArgsConstructor
+public class FileInfoService {
+  private final String pathSeparator = "/";
+  private final FileInfoRepository fileInfoRepository;
+  private final UserService userService;
+  private final school.hei.haapi.service.aws.FileService fileService;
+
+  public FileInfo uploadFile(
+      String fileName, FileType fileType, String userId, MultipartFile fileToUpload) {
+    User student = userService.findById(userId);
+    String endPath = fileType + pathSeparator + fileName;
+    // STUDENT/ref/TRANSCRIPT | DOCUMENT | OTHER/fileName
+    String fileKeyUrl = getFormattedBucketKey(student, endPath);
+    FileInfo savingFileInfo =
```

Comment:
why do you use "ing" file names, we mostly use past participle verbs or infinitive